### DISCUSSION TITLE
Animate backward/forward visits

### DISCUSSION THREAD

**Discussion by andresclua** [swup/swup#920] [Help] [answered]
### Describe your problem 🧐

Page transition is not happening when I’m navigating using the prev/next browser arrows, only happens if i click on a link

https://t2197638.p.clickup-attachments.com/t2197638/6f91ea51-d8d0-433f-bd50-0689d70ba9e1/RPReplay_Final1720079256.MP4?view=open

this is how my setup is built.

`
class Core {
    constructor(payload) {
        this.swup = new Swup({
            linkSelector: "a[href], area[href], svg a[*|href]",
            plugins: [
                new SwupHeadPlugin(),
                new SwupScriptsPlugin({
                    head: true,
                    body: true,
                }),
                new SwupDebugPlugin({
                    globalInstance: true,
                }),
                new SwupJsPlugin(transitionOptions),
            ],
        });

    }
    
    ``

### What is your question? 🙋‍♀️

How could I prevent that behaviour and make same transition when I click native buttons (prev,next) from the browser ?

### URL to minimal reproduction 🔗

https://sei2024dev.wpenginepowered.com/

### Checked all these? 📚

- [X] I have read and searched the [official docs](https://swup.js.org)
- [X] I have checked [discussions](https://github.com/orgs/swup/discussions) and [existing issues](https://github.com/issues?q=is:issue+org:swup) for answers to my question
- [X] I agree to follow this project's [Code of Conduct](https://github.com/swup/.github/blob/master/CODE_OF_CONDUCT.md)
- [X] I have provided all necessary information to the best of my knowledge

**Comment by hirasso** [comment, marked as answer]
Swup is not animating history visits by default, as that brings various issues with it that could degrade the user experience. If you still decide that you need it, you need to set the option [animateHistoryBrowsing](https://swup.js.org/options/#animatehistorybrowsing) to `true`. Please make sure to read the warning on the docs before you decide. To restore the restoration of your scroll positions, you should have a look at [@swup/scroll-plugin](https://swup.js.org/plugins/scroll-plugin/).
